### ISSUE TITLE
[BUG] Element is not opened in fullscreen mode ( chrome / chromium ) 

### ISSUE THREAD

**Issue by PiotrSF** [microsoft/playwright#27598]
### System info
- Playwright Version: 1.39.0
- Operating System: macOS Ventura 13.5 
- Browser: Chromium
- Other info:

### Source code / Test code 

test.describe(`Fullscreen tests`, () => {

    test(`#1 Should open Fullscreen`, async ({ page })=> {

        await page.goto(`https://pktestblog1.blogspot.com/p/fullscreen-page.html`);
        await page.locator('#openFullscreen').click();
    })
})

HTML code for simple FS page is in .pdf in attachments 

**Config file**

playwright.config.ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  projects: [
 {
      name: 'Google Chrome',
      use: {
            ...devices['Desktop Chrome'], channel: 'chrome',
            launchOptions: {
            chromiumSandbox: true,
            ignoreDefaultArgs: [`--hide-scrollbars`],
            args: ['--disable-web-security'],
            },
            permissions: ["clipboard-read"],
     },
    },

**Expected**

Element should be opened in FS mode, works for firefox / webkit : 
<img width="1250" alt="FS in firefox" src="https://github.com/microsoft/playwright/assets/115999141/533474e6-9403-46c4-a3d3-a54cba1e6912">

**Actual**

Button is found and clicked but element is not opened in fullscreen mode 
<img width="1024" alt="no FS in chrome" src="https://github.com/microsoft/playwright/assets/115999141/f7bad05c-931c-4263-be52-54b675c5aa3b">

**Note**
Fullscreen functionality for web element is crucial for our product and in our tests, thanks in advance. 

[fullscreen page.pdf](https://github.com/microsoft/playwright/files/12893911/fullscreen.page.pdf)

**Comment by viraxslot** [comment]
Hi, I believe the issue might be related to the screenshot being taken too quickly, which is why the full-screen mode isn't visible. I made some slight changes to your script, and it works fine for me.

```typescript
test.describe.only(`Fullscreen tests`, () => {
  test(`#1 Should open Fullscreen`, async ({ page }) => {
    const title = "Fullscreen page";
    await page.goto(`https://pktestblog1.blogspot.com/p/fullscreen-page.html`);
    await expect(page.getByText(title)).toBeVisible();
    await page.locator("#openFullscreen").click();
    
    // for checking
    await page.waitForTimeout(500);
    await page.screenshot({ path: "full-screen.png" });
  });
});
```

Of course it's a bad practise to wait for a timeout. So you can add some hidden element that will appear/diappear in the full-screen mode and wait for it.

The result of the above script:
<img width="1145" alt="image" src="https://github.com/microsoft/playwright/assets/20737672/aaa8f774-8ee9-43a4-909c-1e27d87e397e">

**Comment by kmlynski** [comment]
Hi @viraxslot 
Your solution seems to not be working on chromium still. Here's the link to quick reproduction:
https://try.playwright.tech/?l=javascript&s=pzvu6ds

**Comment by PiotrSF** [comment]
Hey @viraxslot,

thanks for the comment but I was checking with time outs also before I created an issue but they don't help for chromium in anyway.

**Comment by yury-s** [comment]
> Hi @viraxslot Your solution seems to not be working on chromium still. Here's the link to quick reproduction: https://try.playwright.tech/?l=javascript&s=pzvu6ds

The test will pass in headed if you move waitForTimeout to line 17:
<img width="459" alt="image" src="https://github.com/microsoft/playwright/assets/9798949/a56819cd-848b-4aba-b136-c325621e3e24">

**Comment by yury-s** [comment]
> Hey @viraxslot,
> 
> thanks for the comment but I was checking with time outs also before I created an issue but they don't help for chromium in anyway.

The element does not get into full screen mode in headless browser today. I headed it should be working fine if yo wait long enough. See also https://github.com/microsoft/playwright/issues/1086

**Comment by yury-s** [comment]
Filed upstream bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1493760

Meanwhile you can run with `--headless=new` as a workaround:

```js
import { test, expect } from '@playwright/test';

test.use({
  launchOptions: {
    args: ['--headless=new']
  }
})

test(`simple test`, async ({ page }) => {
  await page.setContent(`
    <script>
    function openFullscreen(event) {
      event.target.requestFullscreen();
    }
    </script>
    <div style="width: 500; height: 500; background-color: green" onclick='openFullscreen(event)'></div>
    `);
  await page.locator('div').click();
  await page.waitForTimeout(1000);
  await page.screenshot({ path: "full-screen.png" });
});
```

**Comment by PiotrSF** [comment]
Thanks @yury-s,  `args: ['--headless=new'] ` works great!

**Comment by yury-s** [comment]
> Filed upstream bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1493760

The upstream issue was fixed and the fix should make it to Playwright in 1.41.

**Comment by jaktestowac** [comment]
@yury-s can you confirm that it is part of Playwright 1.41?

**Comment by PiotrSF** [comment]
Yup, I'm looking forward for this fix because `args: ['--headless=new']` is not working in some cases anyway :/

**Comment by PiotrSF** [comment]
Was this fixed removed from 1.41 version ? :/

**Comment by carolhmj** [comment]
I'm seeing the same problem on 1.42, would love an update. I can provide a repro.
